server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -207,36 +207,40 @@ private void extensionsDiscovery() throws IOException {
                 throw new IOException("Could not read from extensions.yml", e);
             }
             for (Extension extension : extensions) {
-                try {
-                    DiscoveryExtension discoveryExtension = new DiscoveryExtension(
-                        extension.getName(),
-                        extension.getUniqueId(),
-                        // placeholder for ephemeral id, will change with POC discovery
-                        extension.getUniqueId(),
-                        extension.getHostName(),
-                        extension.getHostAddress(),
-                        new TransportAddress(InetAddress.getByName(extension.getHostAddress()), Integer.parseInt(extension.getPort())),
-                        new HashMap<String, String>(),
-                        Version.fromString(extension.getOpensearchVersion()),
-                        new PluginInfo(
+                if (extensionIdMap.containsKey(extension.getUniqueId())) {
+                    logger.info("Duplicate uniqueId " + extension.getUniqueId() + ". Did not load extension: " + extension);
+                } else {
+                    try {
+                        DiscoveryExtension discoveryExtension = new DiscoveryExtension(
                             extension.getName(),
-                            extension.getDescription(),
-                            extension.getVersion(),
```

Comment:
Why are we not considering `Description` and `Version` of the extension?

Reply:
Okay I see. We are getting it from `PluginInfo`. Ignore my comment here

Reply:
For clarity (and for other reviewers), there are no changes to this code block other than the indentation caused by the `else`.